src/sinol_make/commands/run/__init__.py

```diff
@@ -725,9 +748,21 @@ def convert_to_expected(results):
                 for group in config_expected_scores[solution]["expected"]:
                     used_groups.add(group)
         else:
-            for solution in results.keys():
-                for group in results[solution].keys():
-                    used_groups.add(group)
+            used_groups = self.get_whole_groups()
+
+            solutions_to_delete = []
+            for solution in new_expected_scores.keys():
+                groups_to_remove = []
+                for group in new_expected_scores[solution]["expected"]:
+                    if group not in used_groups:
+                        groups_to_remove.append(group)
+                for group in groups_to_remove:
+                    del new_expected_scores[solution]["expected"][group]
+                if len(new_expected_scores[solution]["expected"]) == 0:
+                    solutions_to_delete.append(solution)
+            for solution in solutions_to_delete:
+                del new_expected_scores[solution]
```

Comment:
It's hard to tell what this code does at a first glance. Do you have some ideas how that can be improved?

Reply:
I added some comments, is it better now?